src/node/stage/approved.rs

```diff
@@ -678,225 +729,268 @@ impl Approved {
+    fn poll_churn_event_backlog(&mut self, core: &mut Core) -> Result<bool> {
+        if !self.can_poll_churn() {
+            return Ok(false);
+        }
+
+        if let Some(event) = self.shared_state.churn_event_backlog.pop_back() {
+            trace!(
+                "churn backlog poll {:?}, Others: {:?}",
+                event,
+                self.shared_state.churn_event_backlog
+            );
+
+            self.handle_accumulated_event(core, event, AccumulatingProof::default())?;
+            Ok(true)
+        } else {
+            Ok(false)
+        }
+    }
+
+    fn check_ready_or_backlog_churn_event(
+        &mut self,
+        event: AccumulatingEvent,
+    ) -> Option<AccumulatingEvent> {
+        let start_churn_event = match &event {
+            AccumulatingEvent::Online(_)
+            | AccumulatingEvent::Offline(_)
+            | AccumulatingEvent::Relocate(_) => true,
+            _ => false,
+        };
+
+        if start_churn_event && !self.can_poll_churn() {
+            trace!(
+                "churn backlog {:?}, Other: {:?}",
+                event,
+                self.shared_state.churn_event_backlog
+            );
+            self.shared_state.churn_event_backlog.push_front(event);
+            return None;
+        }
+
+        Some(event)
+    }
+
+    // Generate a new section info based on the current set of members and vote for it if it
+    // changed.
+    fn promote_and_demote_elders(&mut self, core: &Core) -> Result<bool> {
+        if !self.members_changed || !self.can_poll_churn() {
+            // Nothing changed that could impact elder set, or we cannot process it yet.
+            return Ok(false);
+        }
+
+        self.members_changed = false;
+
+        let new_infos = if let Some(new_infos) = self
+            .shared_state
+            .promote_and_demote_elders(&core.network_params, core.name())?
+        {
+            self.churn_in_progress = true;
+            new_infos
+        } else {
+            self.churn_in_progress = false;
+            return Ok(false);
+        };
+
+        if !self.is_our_elder(core.id()) {
+            return Ok(true);
+        }
+
+        for info in new_infos {
+            let participants: BTreeSet<_> = info.member_ids().copied().collect();
+            let _ = self.dkg_cache.insert(participants.clone(), info);
+            self.vote_for_event(AccumulatingEvent::StartDkg(participants));
+        }
+
+        Ok(true)
+    }
+
+    /// Polls and handles the next scheduled relocation, if any.
+    fn poll_relocation(&mut self, our_id: &PublicId) -> bool {
+        // Delay relocation until no additional churn is in progress.
+        if !self.can_poll_churn() {
+            return false;
+        }
+
+        if let Some(details) = self.shared_state.poll_relocation() {
+            if self.is_our_elder(our_id) {
+                self.vote_for_relocate_prepare(details, INITIAL_RELOCATE_COOL_DOWN_COUNT_DOWN);
```

Comment:
I'd keep the phrase "cool down" because we've been using it for this specific thing. It's supposed to mean the number of iterations (rounds/steps/...) of the relocation cool-down period. So maybe just `RELOCATE_COOL_DOWN_STEPS`  /`RELOCATE_COOL_DOWN_ITERATIONS` / `RELOCATE_COOL_DOWN_ROUNDS`?

Reply:
Ah ok, all those are better. So is any of those (rounds/steps/iterations) used more often to refer to this? I would go with that one then, or else any of the shorter ones.